### PR TITLE
Alex/fix test estimated score

### PR DESCRIPTION
**Pull request by choroshin**

### CODE REVIEW

File: test/spec/estimatedScore/estimatedScoreEventsHandlerSrv.spec.js [hunk 1]
```diff
@@ -64,41 +67,40 @@ describe('testing service "EstimatedScoreEventsHandlerSrv":', function () {
...
-        $rootScope.$digest();
+        // $rootScope.$broadcast(exerciseEventsConst.section.FINISH, section, resultsMock, exam);
+        actions.estimatedScoreEventsHandlerSrv.calculateRawScore(exerciseEventsConst, section, resultsMock, exam);
+         $rootScope.$digest();
         scoresArr = actions.getEstimatedScoresFromDb(section.subjectId);
         expect(scoresArr.length).toEqual(1);
     });
 
-    it('when writing section is completed in diagnostic test then raw score object should be calculated accordingly', function () {
+    xit('when writing section is completed in diagnostic test then raw score object should be calculated accordingly', function () {
```
Comment (devzinkerz) [inline test/spec/estimatedScore/estimatedScoreEventsHandlerSrv.spec.js]: remove x

File: test/jsonFixtures/content.json [hunk 2]
```diff
@@ -9,6 +9,7 @@
                 "name": "Adaptive Diagnostic English",
                 "examId": 47,
                 "subjectId": 5,
+                "categoryId": 164,
```
Comment (devzinkerz) [inline test/jsonFixtures/content.json]: check

File: src/components/estimatedScore/services/estimatedScoreEventsHandlerSrv.js [hunk 3]
```diff
@@ -96,23 +96,49 @@
...
-                        var question = questionsMap[result.questionId];
-                        if (angular.isUndefined(question)) {
-                            $log.error('EstimatedScoreEventsHandler: question for result is missing',
-                                'section id: ', section.id,
-                                'result index: ', i
-                            );
-                        } else {
-                            score += _getDiagnosticQuestionPoints(question, result);
-                        }
+                        var scorePromise = $q(function (resolve, reject) {
```
Comment (devzinkerz) [inline src/components/estimatedScore/services/estimatedScoreEventsHandlerSrv.js]: why this syntax ?

File: dist/estimatedScore/estimatedScore.js [hunk 4]
```diff
@@ -112,23 +113,49 @@
...
+                                    subjectId1Prom,
+                                    subjectId2Prom
+                                ]).then(function (subjectIds) {
+                                    angular.forEach(subjectIds, function (subjectId) {
+                                        if (angular.isNumber(subjectId)) {
+                                            if (angular.isUndefined(scores[subjectId])) {
+                                                scores[subjectId] = 0;
+                                            }
+                                            scores[subjectId] += _getDiagnosticQuestionPoints(question, result);
+                                        }
```
Comment (devzinkerz) [inline dist/estimatedScore/estimatedScore.js]: why it is always has to getDiangostic?

File: dist/estimatedScore/estimatedScore.js [hunk 5]
```diff
@@ -112,23 +113,49 @@
...
+                                }); // then
+                            }
+                        });
+                        scoresPromises.push(scorePromise);
+                    });
+
+                    $q.all(scoresPromises).then(function () {
+                        var subjectIds = Object.keys(scores);
+                        subjectIds.forEach(function (subjectId) {
+                            EstimatedScoreSrv.setDiagnosticSectionScore(scores[subjectId], ExerciseTypeEnum.SECTION.enum, subjectId, section.id);
```
Comment (devzinkerz) [inline dist/estimatedScore/estimatedScore.js]: you know that subjectId here is string.....is it ok ?

File: test/jsonFixtures/content.json (hunk 2 above)
Comment (choroshin) [inline test/jsonFixtures/content.json]: removed ` "categoryId": 164,` , run dist, tests passed successfully.

File: src/components/estimatedScore/services/estimatedScoreEventsHandlerSrv.js (hunk 3 above)
Comment (choroshin) [inline src/components/estimatedScore/services/estimatedScoreEventsHandlerSrv.js]: changed

File: test/spec/estimatedScore/estimatedScoreEventsHandlerSrv.spec.js (hunk 1 above)
Comment (choroshin) [inline test/spec/estimatedScore/estimatedScoreEventsHandlerSrv.spec.js]: removed

File: dist/estimatedScore/estimatedScore.js (hunk 5 above)
Comment (choroshin) [inline dist/estimatedScore/estimatedScore.js]: yes, seems it doesn't have any effect.

Comment (assafshp) [inline dist/estimatedScore/estimatedScore.js]: ok

**Pull request merged**